From schmitz@opal.biophys.uni-duesseldorf.de Fri Jun 27 10:42:06 2008
Date: Fri, 27 Jun 2008 10:40:43 +0200 (CEST)
From: Michael Schmitz <schmitz@opal.biophys.uni-duesseldorf.de>
To: Michael Schmitz <schmitz@opal.biophys.uni-duesseldorf.de>
Cc: Stephen R Marenka <stephen@marenka.net>, linux-m68k@vger.kernel.org, Geert Uytterhoeven <geert@linux-m68k.org>
Subject: Re: ide_release_lock: bug

Hi,

The following patch does, indeed, fix the ide_release_lock imbalance. No
more warnings.

---
 drivers/ide/ide-io.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/ide/ide-io.c
+++ b/drivers/ide/ide-io.c
@@ -982,14 +982,13 @@ static void ide_do_request (ide_hwgroup_
 	ide_startstop_t	startstop;
 	int             loops = 0;
 
-	/* for atari only: POSSIBLY BROKEN HERE(?) */
-	ide_get_lock(ide_intr, hwgroup);
-
 	/* caller must own ide_lock */
 	BUG_ON(!irqs_disabled());
 
 	while (!hwgroup->busy) {
 		hwgroup->busy = 1;
+		/* for atari only */
+		ide_get_lock(ide_intr, hwgroup);
 		drive = choose_drive(hwgroup);
 		if (drive == NULL) {
 			int sleeping = 0;
